lib/cartopy/mpl/gridliner.py

```diff
@@ -411,16 +578,48 @@ def _find_midpoints(self, lim, ticks):
                      self._round(np.percentile(lim, uq), cent))
         return midpoints
 
+    def _draw_this_label(self, xylabel, loc):
+        """Should I draw this kind of label here?"""
+        draw_labels = getattr(self, loc+'_labels')
+        if not draw_labels:
+            return False
+        if (isinstance(draw_labels, str) and draw_labels != xylabel):
+            return False
+        if ((xylabel == "x" and loc in ["left", "right"]) or
+                (xylabel == "y" and loc in ["top", "bottom"])):
+            return False
```

Comment:
Here is suggestion:
```suggestion
        draw_labels = getattr(self, loc+'_labels')
        if draw_labels is True and loc != 'geo':
            draw_labels = (
                "x" if loc in ["top", "bottom"] else "y")
        if not draw_labels:
            return False
        if (isinstance(draw_labels, str) and draw_labels != xylabel):
            return False
        return True
```

Reply:
:+1: Might want to put a comment or two in to discuss the logic going through here.